sklego/preprocessing.py

```diff
@@ -100,13 +100,13 @@ def transform(self, X):
         :param X: pandas dataframe to select dtypes for
         """
         check_is_fitted(self, ['type_columns_', 'X_dtypes_', 'feature_names_'])
+        self._check_X_for_type(X)
```

Comment:
Originally  _check_X_for_type(X) is called after checking dtypes. By this if we call transform method on numpy array then it raise
`AttributeError: 'numpy.ndarray' object has no attribute dtypes`

but _check_X_for_type(X) should be called before checking dtypes, and then calling transform on numpy array raises,
`TypeError: Provided variable X is not of type pandas.DataFrame`

which is the suitable and correct error message to user.